### PR TITLE
8265489: Stress test times out because of long ObjectSynchronizer::monitors_iterate(...) operation

### PR DESCRIPTION
**Pull request by lmesnik**
monitors_iterate make several checks which often are true before filter monitor by a thread. It might take a lot of time when there are a lot of threads. So it makes sense to first check thread and only then other conditions.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265489](https://bugs.openjdk.java.net/browse/JDK-8265489): Stress test times out because of long ObjectSynchronizer::monitors_iterate(...) operation

### Reviewers
 * [Daniel D. Daugherty](https://openjdk.java.net/census#dcubed) (@dcubed-ojdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5194/head:pull/5194` \
`$ git checkout pull/5194`

Update a local copy of the PR: \
`$ git checkout pull/5194` \
`$ git pull https://git.openjdk.java.net/jdk pull/5194/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5194`

View PR using the GUI difftool: \
`$ git pr show -t 5194`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5194.diff">https://git.openjdk.java.net/jdk/pull/5194.diff</a>

</details>

### CODE REVIEW

**Review by dcubed-ojdk — CHANGES_REQUESTED** [review]

File: src/hotspot/share/runtime/synchronizer.cpp [hunk 1]
```diff
@@ -972,10 +972,13 @@ JavaThread* ObjectSynchronizer::get_lock_owner(ThreadsList * t_list, Handle h_ob
...
 // Visitors ...
 
-void ObjectSynchronizer::monitors_iterate(MonitorClosure* closure) {
+void ObjectSynchronizer::monitors_iterate(MonitorClosure* closure, JavaThread* thread) {
   MonitorList::Iterator iter = _in_use_list.iterator();
   while (iter.has_next()) {
     ObjectMonitor* mid = iter.next();
+    if (mid->owner() != thread) {
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (dcubed-ojdk) [reply src/hotspot/share/runtime/synchronizer.cpp]: Thanks for resolving the above comment.

**Review by dcubed-ojdk — APPROVED** [review]
Moving the thread check from the closure's do_monitor() call
into monitors_iterate() as early as possible is a good idea.

Do you have any measurements to show how much this helps?
I'm okay if you don't and I'd be happy waiting to see if it makes
a difference with some of those Tier8 timeouts...

**Comment by lmesnik** [comment]
I run stress tests several times on linux-x64 with ParallelGC and haven't seen timeout anymore.

**Pull request closed without merging**